x-pack/plugins/security_solution/public/management/cypress/e2e/response_actions/response_console.cy.ts

```diff
@@ -228,48 +242,163 @@ describe('Response console', { tags: ['@ess', '@serverless', '@brokenInServerles
+        waitForCommandToBeExecuted('execute');
+      });
+
+      it('"upload --file" - should upload a file', () => {
+        waitForEndpointListPageToBeLoaded(createdHost.hostname);
+        openResponseConsoleFromEndpointList();
+        inputConsoleCommand(`upload --file`);
+        cy.getByTestSubj('console-arg-file-picker').selectFile(
+          {
+            contents: Cypress.Buffer.from('upload file content here!'),
+            fileName: 'upload_file.txt',
+            lastModified: Date.now(),
+          },
+          { force: true }
         );
-        cy.contains('Click here to download').click();
-        const downloadsFolder = Cypress.config('downloadsFolder');
-        cy.readFile(`${downloadsFolder}/upload.zip`);
+        submitCommand();
+        waitForCommandToBeExecuted('upload');
+      });
+    });
 
-        cy.task('uploadFileToEndpoint', {
-          hostname: createdHost.hostname,
-          srcPath: `${downloadsFolder}/upload.zip`,
-          destPath: `${homeFilePath}/upload.zip`,
+    describe('From Alerts', function () {
+      let ruleId: string;
+      let ruleName: string;
+
+      before(() => {
+        loadRule(
+          { query: `agent.name: ${createdHost.hostname} and agent.type: endpoint` },
+          false
+        ).then((data) => {
+          ruleId = data.id;
+          ruleName = data.name;
         });
+      });
 
-        cy.task('readZippedFileContentOnEndpoint', {
-          hostname: createdHost.hostname,
-          path: `${homeFilePath}/upload.zip`,
-          password: 'elastic',
-        }).then((unzippedFileContent) => {
-          expect(unzippedFileContent).to.equal(fileContent);
-        });
+      after(() => {
+        if (ruleId) {
+          cleanupRule(ruleId);
+        }
+      });
+
+      it('should open responder from alert details flyout', () => {
+        waitForEndpointListPageToBeLoaded(createdHost.hostname);
+        toggleRuleOffAndOn(ruleName);
+        visitRuleAlerts(ruleName);
+        closeAllToasts();
+        openAlertDetailsView();
+
+        openResponderFromEndpointAlertDetails();
+        cy.getByTestSubj('consolePageOverlay-layout-titleHolder').should('exist');
+      });
+
+      it('should open responder from timeline view alert details flyout', () => {
+        waitForEndpointListPageToBeLoaded(createdHost.hostname);
+        toggleRuleOffAndOn(ruleName);
+        visitRuleAlerts(ruleName);
+        closeAllToasts();
+
+        openInvestigateInTimelineView();
+        openAlertDetailsView();
+        openResponderFromEndpointAlertDetails();
+        verifyResponderIsOpen();
       });
     });
 
-    it('"execute --command" - should execute a command', () => {
-      waitForEndpointListPageToBeLoaded(createdHost.hostname);
-      openResponseConsoleFromEndpointList();
-      inputConsoleCommand(`execute --command "ls -al ${homeFilePath}"`);
-      submitCommand();
-      waitForCommandToBeExecuted('execute');
+    describe('From Cases', () => {
```

Comment:
could we extract each `describe` to a separate file to avoid long-running single specs?

Reply:
10ea4c96d9e6409cd621e698d34d1cf780a9ccd1